### PR TITLE
#160371227 Send notification for approved request 

### PR DESCRIPTION
**Pull request by Nedson202**
#### What does this PR do?
- Sends notification to the requesters if their notification has been approved by their manager.

#### Description of Task to be completed?
- create an endpoint to retrieve a user's notification
- Add  notify method to approval controller to send a notification to requester after their request has been approved

#### How should this be manually tested?
- Clone the repo - git clone https://github.com/andela/travel_tool_back.git
- Enter into the project directory - cd travel_tool_back
- Rollback and run database migrations - yarn db:rollmigrate
- Install dependencies and start the server - make start

#### Any background context you want to provide?
This works alongside this [PR](https://github.com/andela/travel_tool_front/pull/107) on the frontend.

In other to test this feature, you need to update your role on the Users table to that of a manager.
Manager roleId: 50319

#### What are the relevant pivotal tracker stories?
[#160371227](https://www.pivotaltracker.com/story/show/160371227)

#### Screenshots (if appropriate)
N/A
#### Questions:
N/A

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -18,7 +18,9 @@ io.on('connection', (client) => {
   logger(`Socket connection established with ${client.id}`);
   client.on('disconnect', () => logger('Socket disconnected'));
 });
-
+// app.set('socketInstance', io);
+global.io = io;
+// SocketInstance.getInstance(io);
```
Comment (hariclerry) [inline src/index.js]: Please remove the commented out lines of code.

File: src/helpers/socket/newRequest.js [hunk 2]
```diff
@@ -0,0 +1,6 @@
+export default function newRequest(client, io) {
+  client.on('createRequest', (data) => {
+    console.log('+++ data from newRequest connection +++ ', data, client.id);
```
Comment (hariclerry) [inline src/helpers/socket/newRequest.js]: Please remove this console.log() line.

**Comment by Nedson202** [comment]
Thanks, @hariclerry for the feedback. Can you please confirm that it has been resolved

**Comment by Nedson202** [comment]
Let me look into this and fix it @Collins33. Thanks for pointing this out.

**Comment by samuelkubai** [comment]
@Nedson202 I need this PR picking up more pace. It's a big blocker to deployments.

**Comment by Nedson202** [comment]
Sorry for the delay @samuelkubai. Let me push more for reviewers.

File: src/modules/requests/__tests__/requests.test.js [hunk 3]
```diff
@@ -5,13 +5,26 @@ import models from '../../../database/models';
 import app from '../../../app';
 import testRequests from './mocks/mockData';
 import Utils from '../../../helpers/Utils';
+import { role } from '../../userRole/__tests__/mocks/mockData'
 
 const request = supertest;
 
+const dataToEmit = {
```
Comment (amarachukwu-agbo) [inline src/modules/requests/__tests__/requests.test.js]: Is this needful, if I remove it, my tests still pass

File: src/modules/approvals/ApprovalsController.js [hunk 4]
```diff
@@ -116,6 +122,28 @@ class ApprovalsController {
...
+        senderName: name,
+        senderImage: picture,
+        recipientId: userId,
+        notificationType: 'general',
+        requestId: id,
+        message: 'approved your request',
+        notificationLink: `/requests/${id}`,
+      };
+
+      return status === 'Approved'
```
Comment (amarachukwu-agbo) [inline src/modules/approvals/ApprovalsController.js]: Is the user only notified when a request is approved?

File: src/modules/requests/__tests__/requests.test.js (hunk 3 above)
Comment (Nedson202) [reply src/modules/requests/__tests__/requests.test.js]: Let me look into this.

Comment (Nedson202) [reply src/modules/requests/__tests__/requests.test.js]: This has been resolved @amarachukwu-agbo.

File: src/modules/approvals/ApprovalsController.js (hunk 4 above)
Comment (Nedson202) [reply src/modules/approvals/ApprovalsController.js]: That's correct. This was clarified by Olamide on the story thread.

**Comment by samuelkubai** [comment]
@hariclerry please explicitly approve or request for changes on the PR.

**Comment by samuelkubai** [comment]
@Nedson202 isn't this PR ready for a TC?

**Comment by Nedson202** [comment]
@samuelkubai Am currently with him. He's looking into this.

**Comment by samuelkubai** [comment]
@Nedson202 why isn't the label properly updated?

**Comment by babadee001** [comment]
LGTM @samuelkubai

**Pull request merged**